hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -203,13 +205,17 @@ private void balance() {
           } catch (InterruptedException e) {
             LOG.info("Container Balancer was interrupted while waiting for" +
                 " next iteration.");
-            stop();
+            if (!isBalancerRunning()) {
+              stop();
+            }
```

Comment:
the only `interrupt()` is called in `stop()` , so if we reach here, `isBalancerRunning()` will always return false. 
we can remove these code , and this also avoids  dead lock